packages/x-data-grid-premium/src/tests/rowGrouping.DataGridPremium.test.tsx

```diff
@@ -1605,7 +1605,7 @@ describe('<DataGridPremium /> - Row grouping', () => {
       expect(getColumnValues(1)).to.deep.equal(['', '0', '3', '', '1', '4', '', '2']);
     });
 
-    it('should not use valueGetter to group the rows when defined', () => {
+    it('should use valueGetter to group the rows when defined', () => {
```

Comment:
It turned out that the current behavior was intentional.
I disagree with this, but maybe I don't see the full picture.
I know it's been 3 years 😅, but still – @flaviendelangle if you remember any additional details about this decision – please let me know 🙂

Reply:
We discussed it some time ago with @michelengelen and the problem is that I don't remember the reason why we did that :laughing: 
But I remember we did it on purpose (maybe a bad one though).